pkg/controller/operators/catalog/operator.go

```diff
@@ -569,3 +454,340 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 
 	return nil
 }
+
+// DependencyResolver defines how a something that resolves dependencies (CSVs, CRDs, etc...)
+// should behave
+type DependencyResolver interface {
```

Comment:
Prefer using function interfaces for more flexibility. Also, can we clean up the signatures (consistent parameter types and ordering, return values, no side effects, etc)?